cpp/DEVELOPER_GUIDE.md

```diff
@@ -9,6 +9,74 @@ cuML is thread safe so its functions can be called from multiple host threads if
 
 The implementation of cuML is single threaded.
 
+## Managing state
```

Comment:
I think this strongly relates to the existing `## C APIs` section. So this should either be a subsection of `## C APIs` should be merged into this.

Reply:
Hmm.. I think the title of the section can cause confusion among the readers. This section is mostly to deal with how to expose state-related info from the __C++__ interface. Let me rename the section properly.